src/Symfony/Component/Form/Extension/Core/Type/LanguageType.php

```diff
@@ -42,16 +44,38 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults([
             'choice_loader' => function (Options $options) {
                 $choiceTranslationLocale = $options['choice_translation_locale'];
-
-                return new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale) {
-                    return array_flip(Languages::getNames($choiceTranslationLocale));
+                $choiceSelfTranslation = $options['choice_self_translation'];
+
+                return new IntlCallbackChoiceLoader(static function () use ($choiceTranslationLocale, $choiceSelfTranslation) {
+                    if ($choiceSelfTranslation) {
+                        foreach (Languages::getNames() as $languageCode => $languageName) {
+                            try {
+                                $languagesList[$languageCode] = Languages::getName($languageCode, $languageCode);
+                            } catch (MissingResourceException $e) {
```

Comment:
should we avoid silent ignoring and always have a consistent list? thus using the original name here.